doc/classes/vector.adoc

```diff
@@ -0,0 +1,264 @@
+
+=== `vector.to_string(v) -> string`
+* `v`: `vector`
+
+Returns a `string` representation of `v` in the format: `(v.x, v.y, v.z)`.
+
+Each component is formatted with the printf-style `%g` flag (either floating-point or scientific notation, whichever is shorter).
+
+=== `vector.equals(a, b) -> bool`
+* `a`, `b`: `vector`
+
+Returns `true` if `a` is equivalent to `b` (all components are the same).
+
+Returns `false` otherwise.
+
+=== `vector.length(v) -> number`
+* `v`: `vector`
+
+Returns the vectorial length (total traveled traveled distance from the origin to the end) of `v`.
+
+The formula for the length is: stem:[sqrt(x^2 + y^2 + z^2)].
+
+=== `vector.normalize(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` which is the normalized form of `v` (the vectorial length is equal to 1).
+Uses `vector.length(v)` to get the vectorial length.
+
+Specifically, if the length is 0, returns a new `vector` with all components being `0`.
+Otherwise, returns `vector.divide(v, length)`.
+
+=== `vector.floor(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where each component of `v` has had `math.floor()` applied to it.
+
+Literally `vector.apply(v, math.floor)`.
+
+=== `vector.round(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where each component of `v` has had `math.round()` applied to it.
+
+Equivalent to `vector.apply(v, math.round)`
+
+=== `vector.apply(v, func) -> vector`
+* `v`: `vector`
+* `func`: `function`
+
+Returns a new `vector` where each component of `v` has had `func()` applied to it.
+
+=== `vector.distance(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` which is equal to the distance between `a` and `b`.
+
+Distance is equal to the scalar (single number) result of stem:[|bar a - bar b|].
+
+=== `vector.direction(pos1, pos2) -> vector`
+* `pos1`, `pos2`: `vector`
+
+Returns a new, normalized `vector` equal to the direction from `pos1` to `pos2`.
+
+=== `vector.angle(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` which is equal to the angle (in radians) between `a` and `b`.
+
+Formula used is stem:[tan^-1(|bar a xx bar b|, bar a * bar b)].
+
+=== `vector.dot(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` equal to the dot product of `a` and `b`.
+
+=== `vector.cross(a, b) -> vector`
+* `a`, `b`: `vector`
+
+Returns a new `vector` which is equal to the cross product of `a` and `b`.
+
+=== `vector.add(a, b) -> vector`
+* `a`: `vector`
+* `b`: `vector` or `number`
+
+If `b` is a `vector`:: Returns a new `vector` where each component of `a` is added to each component of `b`
+If `b` is a `number`:: Returns a new `vector` where each component of `a` is added to `b`
+
+=== `vector.subtract(a, b) -> vector`
+* `a`: `vector`
+* `b`: `vector` or `number`
+
+If `b` is a `vector`:: Returns a new `vector` where each component of `a` is subtracted from each component of `b`
+If `b` is a `number`:: Returns a new `vector` where each component of `a` is subtracted from `b`
```

Comment:
Wrong way around maybe? I'd assume `b` is subtracted from each component of `a`

Reply:
Yes, both are the wrong way around. `b` is subtracted from `a` in both cases.